### ISSUE TITLE
IBM-Granite 

### ISSUE THREAD

**Issue by ALutz273** [ollama/ollama#4209]
Hello,

Very interesting because of software license 

https://github.com/ibm-granite/granite-code-models

**Comment by coder543** [comment]
I'd like to see this as well. Blocked by this issue: https://github.com/ggerganov/llama.cpp/issues/7116

**Comment by kristofer** [comment]
me as well.

**Comment by coder543** [comment]
Support for the 20B and 34B models seems fine now in llama.cpp, but the latest release of ollama is still missing the necessary commits. I [have uploaded](https://github.com/ggerganov/llama.cpp/issues/7116#issuecomment-2120536190) a few GGUFs, but trying to load them in ollama just causes a crash.

**Comment by nurena24** [comment]
Has this been fixed yet?

**Comment by coder543** [comment]
The 20B and 34B models are working great on the 0.1.39 pre-release version of ollama: https://ollama.com/library/granite-code

The underlying llama.cpp library still does not support the smaller Granite models, from what I understand.

**Comment by coder543** [comment]
The PR that adds support for small models was just merged into llama.cpp: https://github.com/ggerganov/llama.cpp/pull/7481

Hopefully we can get those models in ollama soon.

**Comment by sealad886** [comment]
If you download Ollama from GitHub and build it yourself, then make your own modelfile and import into Ollama, then you can run this right now. I've been using it for the past day or so as a coding assist AI...one or two weird things, so I'm not sure which versions I have of everything lining up. Just wanted to let everyone know that it's worth the hype, and you can get in on it too! :)

**Comment by DigitLib** [comment]
Here is gguf Q5_K_M version that works on llama.cpp. Converted and tested today. Hope to see it soon on ollama.
Thank you for this great work!

**Comment by sroecker** [comment]
I think this can be closed now. @jmorganca

**Comment by jmorganca** [comment]
https://ollama.com/library/granite-code @sroecker thanks!

**Comment by ericcurtin** [comment]
I think we messed up the naming here to be honest, I tried a

ollama pull granite

nothing happened. I'm not sure how many would instinctively include the "-code".

What do you think @sroecker ?

Could we claim/migrate to https://ollama.com/library/granite ? @jmorganca would it be possible?

I've never heard anyone call it "Granite code" and I've heard about IBM's granite model a lot.

**Comment by coder543** [comment]
@ericcurtin I’m not a maintainer here, but I’ll point out that [IBM Granite](https://huggingface.co/ibm-granite/granite-7b-base) is largely unrelated to the IBM Granite Code models, and I’ve had the opposite anecdotal experience.

**Comment by ericcurtin** [comment]
@coder543 yes you are correct, I think I have to train myself up on IBM Granite. code is the IBM Granite model specifically designed for code.

**Comment by sroecker** [comment]
> @ericcurtin I’m not a maintainer here, but I’ll point out that [IBM Granite](https://huggingface.co/ibm-granite/granite-7b-base) is largely unrelated to the IBM Granite Code models, and I’ve had the opposite anecdotal experience.

Indeed, I think it's good to keep them separate. The code models themselves are already a bit confusing due to their different archictectures. Also, imagine a few additional non-code models coming out making it even more confusing.
